controlplane/kubeadm/internal/filters_test.go

```diff
@@ -450,7 +478,29 @@ func TestMatchInitOrJoinConfiguration(t *testing.T) {
 				},
 			},
 		}
-		g.Expect(matchInitOrJoinConfiguration(machineConfigs[m.Name], kcp)).To(BeFalse())
+		match, diff, err := matchInitOrJoinConfiguration(machineConfigs[m.Name], kcp)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(match).To(BeFalse())
+		g.Expect(diff).To(BeComparableTo(`&v1beta1.KubeadmConfigSpec{
+    ClusterConfiguration: nil,
+    InitConfiguration: &v1beta1.InitConfiguration{
+      TypeMeta:        {},
+      BootstrapTokens: nil,
+      NodeRegistration: v1beta1.NodeRegistrationOptions{
+-       Name:      "",
++       Name:      "A new name",
+        CRISocket: "",
+        Taints:    nil,
+        ... // 3 identical fields
+      },
+      LocalAPIEndpoint: {},
+      SkipPhases:       nil,
+      Patches:          nil,
+    },
+    JoinConfiguration: nil,
+    Files:             nil,
+    ... // 10 identical fields
```

Comment:
Note: Not sure how good these diffing is, because we want to not have a match.

The diff will change as soon as we add fields (which should in practice normally never happen :-) )

TLDR: ok to do it like this, let the future us handle this if it happens :D

Reply:
In general I like the format that cmp.Diff is producing per default. Theoretically we could implement a custom reporter, but I don't have a good idea how to do it better than they do :)